### ISSUE TITLE
Screen Reader: Many Pages: The fields of the code digits are not announced

### ISSUE THREAD

**Issue by olenamihno** [Expensify/App#77551]
## Action Performed:
Prerequisites:
- the user is not signed in

1. Using iOS VoiceOver, open the Expensify app
2. Enter a valid email address.
3. Swipe to the verification code input fields
4. Note how the field are announced

Other Occurrences:
1. Onboarding - Verify your email
2. On Profile - Verify Email
3. On Security - Merge Accounts
4. On Security - Merge Accounts - Enter Code

## Expected Result:
When the field is focused, it should be announced with the field role and descriptive name, e.g.:
- 'code digit 1 of 6'

## Actual Result:
The fields of the code digits are not announced.

## Area issue was found in:
Sign in

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
Ensure the code fields are implemented as text fields to announce the role and entered value.
https://reactnative.dev/docs/textinput

Add the accessibilityLabel to announce each field descriptively, e.g.:
- 'code digit 1 of 6'
https://reactnative.dev/docs/accessibility#accessibilitylabel

Alternatively, merge the fields as a single input field.

## Environment:
Apple iPhone 11 - iOS 17.5.1, H0T Mobile - Native (No mobile browser)

## Screenshots/Videos

https://github.com/user-attachments/assets/11ebe273-5b91-4378-a3d6-e42df28d6d6c

<img width="828" height="1792" alt="Image" src="https://github.com/user-attachments/assets/269859f7-e330-40a0-8f13-5ae891ae9b26" />

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** No
**Logs:** https://stackoverflow.com/c/expensify/questions/4856  NA
**Notes/Photos/Videos:** Any additional supporting documentation  Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7005410
**Issue reported by:** NA
**Slack conversation:** NA
